control-plane/agents/src/bin/core/volume/snapshot_operations.rs

```diff
@@ -335,88 +348,97 @@ impl OperationGuardArc<VolumeSnapshot> {
         registry: &Registry,
         target_node: N,
     ) -> Result<VolumeSnapshotCreateResult, SvcError> {
-        let mut replica_snap = prep_params.replica_snapshot.1.clone();
-        let replica = &prep_params.replica_snapshot.0;
+        let mut replica_snaps = prep_params.replica_snapshot.1.clone();
+        let replicas = &prep_params.replica_snapshot.0;
         let generic_params = prep_params.parameters.params();
 
-        let replica_id = replica_snap.spec().source_id().replica_id();
+        let replica_ids = replica_snaps
+            .iter()
+            .map(|x| x.spec().source_id().replica_id())
+            .collect::<Vec<_>>();
+
+        let nexus_snap_desc = replica_ids
+            .into_iter()
+            .zip(replica_snaps.clone().into_iter())
+            .map(|(x, y)| CreateNexusSnapReplDescr::new(x, y.spec().uuid().clone()))
+            .collect();
         let response = target_node
             .create_nexus_snapshot(&CreateNexusSnapshot::new(
                 SnapshotParameters::new(target.nexus(), generic_params.clone()),
-                vec![CreateNexusSnapReplDescr::new(
-                    replica_id,
-                    replica_snap.spec().uuid().clone(),
-                )],
+                nexus_snap_desc,
             ))
             .await?;
 
-        if response.skipped.contains(replica_id) || !response.skipped.is_empty() {
+        if !response.skipped.is_empty() {
```

Comment:
Could we just check if skipped contains the replicas which we tried to snap?

Reply:
Currently dataplane skip all the replicas which are marked as skipped from control-plane explicitly. 
If any of the replica which is passed from control-plane and failed to take snapshot, it will be tracked as part of `NexusReplicaSnapshotStatus.status` field in gRPC response from dataplane.